### ISSUE TITLE
apt-get install with variable fails on DL3008

### ISSUE THREAD

**Issue by niroz89** [hadolint/hadolint#363]
- [x] This is a bug report
- [ ] This is a feature request
- [x ] I searched existing issues before opening this one

### Expected behavior
If apt-get install with $buildDeps variable, if the packages in $buildDeps are pinned with version, it should not fail on DL3008
### Actual behavior
fails on DL3008
### Steps to reproduce the behavior
1. Create a Dockerfile like below
```Dockerfile
FROM debian:buster-slim
SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN buildDeps=('ca-certificates=20190110' 'xz-utils=5.2.4-1' 'gnupg=2.2.12-1+deb10u1') \
     && set -ex \
     && apt-get update && apt-get install -y "${buildDeps[@]}" --no-install-recommends \
     && rm -rf /var/lib/apt/lists/* \
     && apt-get purge -y --auto-remove "${buildDeps[@]}"
```
2. Run hadolint on Dockerfile
```bash
hadolint Dockerfile
Dockerfile:4 DL3008 Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
```

**Output of `hadolint --version` 
```bash
hadolint --version
Haskell Dockerfile Linter 1.17.2-no-git
```

**Additional environment details (OS, stack version, etc.)**
MacOS

**Comment by lorenzo** [comment]
Hadolint is not able to interpolate the many variations in which variables can be defined, you can either ignore the rule since you know it is OK, or you can define each package on a separate line, which is my personal preference.

Let me know if you need an example :)

**Comment by niroz89** [comment]
@lorenzo Thank you. I'm already doing ignore now. I have added the $variable since they are build dependencies and deleted after build. Can we still use variable with separate line ?

This is similar to shellcheck https://github.com/koalaman/shellcheck/wiki/SC2086 makefile exception, have to use array.

**Comment by lorenzo** [comment]
This is an excerpt of one of my dockerfiles:

```dockerfile
ENV \
 APACHE_VERSION=2.4.* \
 COMPOSER_ALLOW_SUPERUSER=1 \
 DEBIAN_FRONTEND=noninteractive \
 IMAGEMAGICK_VERSION=8:6.7.7.* \
 LIBGEOS_VERSION=3.4.* 

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

RUN set -ex \
  && apt-get update \
  && apt-get -o Dpkg::Use-Pty=0 -y --force-yes --no-install-recommends install \
      # fundamental requirement, apache is the webserver used
      apache2=${APACHE_VERSION} \
      # required for view-from-seat image uploads
      imagemagick=${IMAGEMAGICK_VERSION}  \
      # fundamental requirements, apache<->php
      libapache2-mod-php7.1=${PHP_VERSION_CORE} \
      # required because reason-missing
      libgeos-dev=${LIBGEOS_VERSION} \
      # required because reason-missing
      librsvg2-bin=${LIBRSVG2_VERSION} 
```

Hope you find this useful!

**Comment by zemanlx** [comment]
Also, it may be useful to use a multistage approach if you need to remove packages after building. In multistage you can install all packages that you need in builder container and then copy build to distribution container. In this scenario, you do not need to purge packages in build image which may save you some precious seconds :wink:

**Comment by niroz89** [comment]
Thank you @lorenzo and @zemanlx . I'm using multistage for the application containers. This one, I'm trying to create a base nodejs dockerfile referring to https://github.com/nodejs/docker-node/blob/master/12/buster-slim/Dockerfile , where they are not using multistage for building nodejs. Sure will test out above two options for the base image. Thanks again, you can close the issue.
